homeassistant/components/awattar/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""The aWATTar integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import AwattarDataUpdateCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up aWATTar from a config entry."""
+
+    coordinator = AwattarDataUpdateCoordinator(hass)
+    try:
+        await coordinator.async_config_entry_first_refresh()
+    except ConfigEntryNotReady:
+        await coordinator.awattar.close()
```

Comment:
Do we need to close here? I saw your are passing a aiohttp session in the coordinator to the client

Reply:
already removed it.